src/sinks/hdfs/config.rs

```diff
@@ -0,0 +1,161 @@
+use codecs::{encoding::Framer, JsonSerializerConfig, NewlineDelimitedEncoderConfig};
+use opendal::{layers::LoggingLayer, services::Hdfs, Operator};
+use vector_config::configurable_component;
+use vector_core::{
+    config::{AcknowledgementsConfig, DataType, Input},
+    sink::VectorSink,
+};
+
+use crate::{
+    codecs::{Encoder, EncodingConfigWithFraming, SinkType},
+    config::{GenerateConfig, SinkConfig, SinkContext},
+    sinks::{
+        opendal_common::*,
+        util::{
+            partitioner::KeyPartitioner, BatchConfig, BulkSizeBasedDefaultBatchSettings,
+            Compression,
+        },
+        Healthcheck,
+    },
+};
+
+/// Configuration for the `hdfs` sink.
```

Comment:
Could we include a link to a the HDFS project site to explain what HDFS is? 

This is the most suitable page I could find, https://hadoop.apache.org/docs/stable/hadoop-project-dist/hadoop-hdfs/HdfsUserGuide.html, but there may be a better one that gives a more useful overview of what HDFS is..

Reply:
Yes, of course. I will add a short description here along with a reference link.